### PR TITLE
added params support

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: src/main/java/com/redislabs/redisgraph/RedisGraph.java [hunk 1]
```diff
@@ -6,15 +6,33 @@
...
+    ResultSet query(String graphId, String query);
+
     /**
      * Execute a Cypher query with arguments
      * @param graphId a graph to perform the query on
      * @param query Cypher query
      * @param args
      * @return a result set
      */
+    @Deprecated
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/RedisGraph.java]: please add @deprecated  to the JDoc pointing to the new method

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java [hunk 2]
```diff
@@ -56,6 +52,7 @@ private static String quoteString(String str){
      * @param args - query arguments
      * @return formatted query
      */
+    @Deprecated
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/impl/Utils.java]: please add @deprecated to the JDoc pointing to the new method

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java [hunk 3]
```diff
@@ -68,6 +65,38 @@ public static String prepareQuery(String query, Object ...args){
...
+        }
+        sb.append(query);
+        return sb.toString();
+    }
+
+    private static String valueToString(Object value) {
+        if(value == null)
+            return "null";
+        Class valueClass = value.getClass();
+        if(valueClass == String.class){
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/impl/Utils.java]: why not instaceof?

File: src/main/java/com/redislabs/redisgraph/impl/api/AbstractRedisGraph.java [hunk 4]
```diff
@@ -29,18 +29,41 @@
      */
     protected abstract ResultSet sendQuery(String graphId, String preparedQuery);
 
+    /**
+     * Execute a Cypher query with arguments
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/impl/api/AbstractRedisGraph.java]: "with arguments"?

File: src/main/java/com/redislabs/redisgraph/RedisGraph.java [hunk 5]
```diff
@@ -6,15 +6,33 @@
 
 public interface RedisGraph extends Closeable {
 
+    /**
+     * Execute a Cypher query with arguments
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/RedisGraph.java]: "with arguments"?

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java [hunk 6]
```diff
@@ -68,6 +65,38 @@ public static String prepareQuery(String query, Object ...args){
...
+     * @param query - query
+     * @param params - query parameters
+     * @return query with parameters header
+     */
+    public static String prepareQuery(String query, Map<String, Object> params){
+        StringBuilder sb = new StringBuilder("CYPHER ");
+        for(Map.Entry<String, Object> entry : params.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            sb.append(key).append("=");
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/impl/Utils.java]: .append('=')

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java [hunk 7]
```diff
@@ -68,6 +65,38 @@ public static String prepareQuery(String query, Object ...args){
...
+            sb.append(valueToString(value));
+            sb.append(" ");
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/impl/Utils.java]: append(' ')

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java [hunk 8]
```diff
@@ -68,6 +65,38 @@ public static String prepareQuery(String query, Object ...args){
...
+
+    private static String valueToString(Object value) {
+        if(value == null)
+            return "null";
+        Class valueClass = value.getClass();
+        if(valueClass == String.class){
+            return quoteString((String) value);
+        }
+        if(valueClass.isArray()){
+            return Arrays.toString((Object[]) value);
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/impl/Utils.java]: Do we need to use quoteString on each value in the array?

File: src/main/java/com/redislabs/redisgraph/impl/api/AbstractRedisGraph.java [hunk 9]
```diff
@@ -29,18 +29,41 @@
...
      * @return a result set
      */
+    @Deprecated
     public ResultSet query(String graphId, String query, Object ...args) {
         String preparedQuery = Utils.prepareQuery(query, args);
         return sendQuery(graphId, preparedQuery);
     }
 
+    /**
+     * Executes a parameterized cypher query.
```
Comment (gkorland) [inline src/main/java/com/redislabs/redisgraph/impl/api/AbstractRedisGraph.java]: with paramets

**Comment by gkorland** [comment]
@DvirDukhan please add negative tests too, like missing params or unused params

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java (hunk 8 above)
Comment (DvirDukhan) [reply src/main/java/com/redislabs/redisgraph/impl/Utils.java]: good catch
changed

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java (hunk 7 above)
Comment (DvirDukhan) [reply src/main/java/com/redislabs/redisgraph/impl/Utils.java]: done

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java (hunk 6 above)
Comment (DvirDukhan) [reply src/main/java/com/redislabs/redisgraph/impl/Utils.java]: done

File: src/main/java/com/redislabs/redisgraph/RedisGraph.java (hunk 5 above)
Comment (DvirDukhan) [reply src/main/java/com/redislabs/redisgraph/RedisGraph.java]: removed

File: src/main/java/com/redislabs/redisgraph/impl/api/AbstractRedisGraph.java (hunk 4 above)
Comment (DvirDukhan) [reply src/main/java/com/redislabs/redisgraph/impl/api/AbstractRedisGraph.java]: removed

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java (hunk 3 above)
Comment (DvirDukhan) [reply src/main/java/com/redislabs/redisgraph/impl/Utils.java]: changed

File: src/main/java/com/redislabs/redisgraph/impl/Utils.java (hunk 2 above)
Comment (DvirDukhan) [reply src/main/java/com/redislabs/redisgraph/impl/Utils.java]: done

File: src/main/java/com/redislabs/redisgraph/RedisGraph.java (hunk 1 above)
Comment (DvirDukhan) [reply src/main/java/com/redislabs/redisgraph/RedisGraph.java]: done

**Pull request merged**